velox/common/memory/tests/MockSharedArbitratorTest.cpp

```diff
@@ -594,43 +594,125 @@ TEST_F(MockSharedArbitrationTest, shrinkPools) {
   struct {
     uint64_t targetBytes;
     uint64_t expectedFreedBytes;
+    bool allowSpill;
+    bool allowAbort;
 
     std::string debugString() const {
       return fmt::format(
-          "targetBytes: {}, expectedFreedBytes: {}",
+          "targetBytes: {}, expectedFreedBytes: {}, allowSpill: {}, allowAbort: {}",
           succinctBytes(targetBytes),
-          succinctBytes(expectedFreedBytes));
+          succinctBytes(expectedFreedBytes),
+          allowSpill,
+          allowAbort);
```

Comment:
Shall we just add a set of flags to control the test expectations? like expectedTaskAbort? instead of checking based on the input condition? Thanks!